apple/RNCWebView.m

```diff
@@ -406,18 +411,46 @@ - (void)didMoveToWindow
 {
   if (self.window != nil && _webView == nil) {
     WKWebViewConfiguration *wkWebViewConfig = [self setUpWkWebViewConfig];
+    NSMutableDictionary *sharedWKWebViewDictionary= [[RNCWKWebViewMapManager sharedManager] sharedWKWebViewDictionary];
+      
+    if ([self shouldReuseWebView]) {
+      WKWebView *webViewForKey = sharedWKWebViewDictionary[_webViewKey];
+      if (webViewForKey != nil) {
+        _webView = webViewForKey;
+        NSMutableDictionary *sharedRNCWebViewDictionary= [[RNCWebViewMapManager sharedManager] sharedRNCWebViewDictionary];
+        RNCWebView *rncWebView = sharedRNCWebViewDictionary[_webViewKey];
+        if (rncWebView != nil) {
+          [self removeWKWebViewFromSuperView:rncWebView];
+        }
+      }
+    }
+    
+    bool reusedWebViewInstance = _webView != nil;
+
+    if (_webView == nil) {
 #if !TARGET_OS_OSX
-    _webView = [[WKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
+      _webView = [[WKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
 #else
-    _webView = [[RNCWKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
+      _webView = [[RNCWKWebView alloc] initWithFrame:self.bounds configuration: wkWebViewConfig];
 #endif // !TARGET_OS_OSX
+    }
+      
+    if (_webViewKey != nil) {
+        NSMutableDictionary *sharedRNCWebViewDictionary= [[RNCWebViewMapManager sharedManager] sharedRNCWebViewDictionary];
+        sharedRNCWebViewDictionary[_webViewKey] = self;
+        if (_webView != nil) {
+          sharedWKWebViewDictionary[_webViewKey] = _webView;
+        }
+    }
     
     [self setBackgroundColor: _savedBackgroundColor];
+    
+    if (!reusedWebViewInstance) {
 #if !TARGET_OS_OSX
-    _webView.scrollView.delegate = self;
+      _webView.scrollView.delegate = self;
 #endif // !TARGET_OS_OSX
-    _webView.UIDelegate = self;
-    _webView.navigationDelegate = self;
+      _webView.UIDelegate = self;
+      _webView.navigationDelegate = self;
```

Comment:
This is curious... this means the previous `RNCWebView`'s delegates would still be in control of the `WKWebView` instance even after it's been reparented to a different `RNCWebView`? What's the intent of guarding this logic with `if (!reusedWebViewInstance)`?

Reply:
That is a great question! You've identified a bug here, and I think we should remove the `if (!resusedWebViewInstance)` check. https://github.com/discord/react-native-webview/pull/21 cc @nealmanaktola 

This was likely a mistake from my first iteration on these changes. If I'm remembering correctly, I was trying to avoid repeating WebView-initialization logic when reusing a `WKWebView` instance with a new `RNCWebView` instance. However, I'm realizing this part of the initialization logic makes the `WKWebView` reference the `RNCWebView`, so it's important that it references the new `RNCWebView` rather than the old one

Reply:
I ran into this issue today when I was trying to make a button-click in the web page conditionally load an iframe. Loading the iframe successfully depends on the navigation delegate as I learned from https://stackoverflow.com/questions/41250162/wkwebview-not-loading-content-from-iframe-on-a-webpage , and I realized the navigation delegate wasn't getting reconnected properly when attaching the `WKWebView` to a new `RNCWebView`

Reply:
Fixed here https://github.com/discord/react-native-webview/commit/a4322df5cdbc0ad63dcb983ece35306666c61c7e